src/qt/intro.cpp

```diff
@@ -287,7 +287,7 @@ void Intro::setStatus(int status, const QString &message, quint64 bytesAvailable
         ui->freeSpace->setText("");
     } else {
         m_bytes_available = bytesAvailable;
-        if (ui->prune->isEnabled()) {
+        if (ui->prune->isEnabled() && gArgs.GetIntArg("-prune", -1) != 0) { // prune checkbox is enabled and not manually set to 0
```

Comment:
Could comment be dropped as it says exactly what code does?

Also I doubt a bit about using `-1` as the default value, because `GetIntArg("-prune", 0)` is used across the whole code base. It could be worked out as
```suggestion
        if (ui->prune->isEnabled() && !(gArgs.IsArgSet("-prune") && gArgs.GetIntArg("-prune", 0) == 0)) {
```

Reply:
Thanks for the improvements. You are right.